packages/@aws-cdk/aws-ecr/lib/repository-ref.ts

```diff
@@ -0,0 +1,80 @@
+import cdk = require('@aws-cdk/cdk');
+import { RepositoryArn, RepositoryName } from './ecr.generated';
+
+/**
+ * An ECR repository
+ */
+export abstract class RepositoryRef extends cdk.Construct {
+    /**
+     * Import a repository
+     */
+    public static import(parent: cdk.Construct, id: string, props: RepositoryRefProps): RepositoryRef {
+        return new ImportedRepository(parent, id, props);
+    }
+
+    /**
+     * The name of the repository
+     */
+    public abstract readonly repositoryName: RepositoryName;
+
+    /**
+     * The ARN of the repository
+     */
+    public abstract readonly repositoryArn: RepositoryArn;
+
+    public abstract addToResourcePolicy(statement: cdk.PolicyStatement): void;
+
+    /**
+     * Export this repository from the stack
+     */
+    public export(): RepositoryRefProps {
+        return {
+            repositoryName: new RepositoryName(new cdk.Output(this, 'RepositoryName', { value: this.repositoryName }).makeImportValue()),
+            repositoryArn: new RepositoryArn(new cdk.Output(this, 'RepositoryArn', { value: this.repositoryArn }).makeImportValue()),
+        };
+    }
+
+    /**
+     * The URI of the repository, for use in Docker/image references
+     */
+    public get repositoryUri(): RepositoryUri {
+        // Calculate this from the ARN
+        const parts = cdk.Arn.parseToken(this.repositoryArn);
```

Comment:
Would be nice to support `isToken(string)` and then `cdk.Arn.parse(s)` will decide if it wants to parse this as a token or as a string

Reply:
Not sure that makes sense. The outputs of this function are *definitely* not readable from a CDK app (even though you can't tell from the types today, but you should be able to tell that). If it makes a decision at runtime to either return readable types or not, you're bound to make errors against that.